Shoko.Server/Providers/AniDB/UDP/AniDBSocketHandler.cs

```diff
@@ -45,15 +46,18 @@ public async Task<byte[]> Send(byte[] payload)
             {
                 result = await SendUnsafe(payload, result);
             }
-            catch (SocketException e)
+            catch (Exception e) when ((e is SocketException se && se.SocketErrorCode == SocketError.TimedOut) ||
+                                      e is OperationCanceledException)
             {
-                if (e.SocketErrorCode == SocketError.TimedOut)
-                {
-                    // we will retry once and only once.
-                    result = await SendUnsafe(payload, result);
-                }
+                // catches timeouts in both synchronous and asynchronous calls
+                // we will retry once and only once.
+                result = await SendUnsafe(payload, result);
             }
         }
+        catch (TaskCanceledException e) when (e.CancellationToken.IsCancellationRequested)
```

Comment:
The `when` here is unneeded impo. Unless we have another cancellation token source that might also cancel it.

Reply:
I'm also personally in agreement here. The `when` was added due to an overabundance of caution. Will remove it now, not like it'd be too hard to re-add in the future if it does become relevant 👍